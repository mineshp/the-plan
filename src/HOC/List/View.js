import React, { Component } from 'react';
import PropTypes from 'prop-types';
//import { connect } from 'react-redux'
//import { addTodo } from '../actions'
import { Header, Table } from 'semantic-ui-react';
import Row from '../../components/List/ItemRow';

class View extends Component {
    constructor(props) {
        super(props);
        //this.handleChange = this.handleChange.bind(this);
        this.state = {
            listID: props.listID,
            list: {
                items: [],
                headings: []
            }
            //list: DataSource.getBlogPost(props.id)
        };
    }

    componentDidMount() {
        fetch(`/list/${this.state.listID}/view`)
            .then(res => res.json())
            .then(list => this.setState({ list }));
    }

    render() {
        let ItemRow = [];
        const headings = this.state.list.headings;
        const items = this.state.list.items;
        items.map((item) => {
            return ItemRow.push(<Row data={item} key={item.id} listID={this.state.list._id} />);
        });

        let ItemHeadings = [];
        headings.map((heading) => {
            return ItemHeadings.push(
                <Table.HeaderCell key={heading.id}>
                    {heading.name}
                </Table.HeaderCell>
            );
        });

        return (
            <div className="List main">
                <Header as='h3' floated='left' dividing>
                    {this.state.list.listName}
                </Header>
                <Table celled striped>
                    <Table.Header>
                        <Table.Row>
                            {ItemHeadings}
                            <Table.HeaderCell />
                        </Table.Row>
                    </Table.Header>

                    <Table.Body>
                        {ItemRow}
                    </Table.Body>
                </Table>
            </div>
        )
    }
};

View.propTypes = {
  id: PropTypes.string
};

View.defaultProps = {
  id: '1'
};

// //ManageList = connect()(ManageList)

export default View