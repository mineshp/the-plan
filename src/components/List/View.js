import React from 'react'
import ViewList from '../../HOC/List/View.js'
import { Container } from 'semantic-ui-react';

const View = (props) => (
  <Container>
    <ViewList listID={props.match.params.id} />
  </Container>
);

export default View;