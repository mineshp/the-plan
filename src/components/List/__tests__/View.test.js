import React from 'react';
import View from '../View';
import renderer from 'react-test-renderer';

const mockResponse = (status, statusText, response) => {
  return new window.Response(response, {
    status: status,
    statusText: statusText,
    headers: {
      'Content-type': 'application/json'
    }
  });
};

const mockList = {
    items: [
        {
            id: '123',
            name: 'A',
            createdDate: new Date(),
            updatedDate: new Date()
        }
    ],
    headings: []
};

it('renders correctly', () => {
  const viewListIdParam = {
    match: {
        params: {
            id: 1
        }
    }
  };

  window.fetch = jest.fn().mockImplementation(() =>
            Promise.resolve(mockResponse(
                200,
                null,
                JSON.stringify(mockList)))
        );

  const tree = renderer.create(
      <View match={viewListIdParam.match}></View>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});