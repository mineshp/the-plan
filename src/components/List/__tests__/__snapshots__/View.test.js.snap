// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui container"
>
  <div
    className="List main"
  >
    <h3
      className="ui dividing left floated header"
    />
    <table
      className="ui celled striped table"
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
          />
        </tr>
      </thead>
      <tbody
        className=""
      />
    </table>
  </div>
</div>
`;
